File: Android/tuilivekit/src/main/java/com/trtc/tuikit/common/imageloader/ImageOptions.java
package com.trtc.tuikit.common.imageloader;

public class ImageOptions {
    public int placeImage;
    public int errorImage;
    public int roundRadius;
    public boolean isGif;
    public boolean skipMemoryCache;
    public boolean skipDiskCache;

    public ImageOptions(Builder builder) {
        this.placeImage = builder.placeImage;
        this.errorImage = builder.errorImage;
        this.roundRadius = builder.roundRadius;
        this.isGif = builder.isGif;
        this.skipMemoryCache = builder.skipMemoryCache;
        this.skipDiskCache = builder.skipDiskCache;
    }

    public static class Builder {
        private int placeImage = 0;
        private int errorImage = 0;
        private int roundRadius = 0;
        private boolean isGif = false;
        private boolean skipMemoryCache = false;
        private boolean skipDiskCache = false;

        public Builder setPlaceImage(int placeImage) {
            this.placeImage = placeImage;
            return this;
        }

        public Builder setErrorImage(int errorImage) {
            this.errorImage = errorImage;
            return this;
        }

        public Builder setRoundRadius(int roundRadius) {
            this.roundRadius = roundRadius;
            return this;
        }

        public Builder asGif(boolean isGif) {
            this.isGif = isGif;
            return this;
        }

        public ImageOptions build() {
            return new ImageOptions(this);
        }
    }
}

File: Android/common/src/main/java/com/trtc/tuikit/common/imageloader/ImageOptions.kt
package com.trtc.tuikit.common.imageloader

class ImageOptions(
    var placeImage: Int,
    var errorImage: Int,
    var roundRadius: Int,
    var isGif: Boolean,
    var skipMemoryCache: Boolean,
    var skipDiskCache: Boolean
) {
    constructor(builder: Builder) : this(
        builder.placeImage,
        builder.errorImage,
        builder.roundRadius,
        builder.isGif,
        builder.skipMemoryCache,
        builder.skipDiskCache
    )

    class Builder {
        @JvmField
        var placeImage: Int = 0
        @JvmField
        var errorImage: Int = 0
        @JvmField
        var roundRadius: Int = 0
        @JvmField
        var isGif: Boolean = false
        @JvmField
        var skipMemoryCache: Boolean = false
        @JvmField
        var skipDiskCache: Boolean = false

        fun setPlaceImage(placeImage: Int): Builder {
            this.placeImage = placeImage
            return this
        }

        fun setErrorImage(errorImage: Int): Builder {
            this.errorImage = errorImage
            return this
        }

        fun setRoundRadius(roundRadius: Int): Builder {
            this.roundRadius = roundRadius
            return this
        }

        fun asGif(isGif: Boolean): Builder {
            this.isGif = isGif
            return this
        }

        fun build(): ImageOptions {
            return ImageOptions(this)
        }
    }
}